src/ResourceManager/ResourceGraph/Commands.ResourceGraph/Cmdlets/SearchAzureRmGraph.cs

```diff
@@ -0,0 +1,137 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.ResourceGraph.Cmdlets
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Management.Automation;
+
+    using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+    using Microsoft.Azure.Commands.ResourceGraph.Utilities;
+    using Microsoft.Azure.Management.ResourceGraph.Models;
+
+    /// <summary>
+    /// Search-AzureRmGraph cmdlet
+    /// </summary>
+    /// <seealso cref="Microsoft.Azure.Commands.ResourceGraph.Utilities.ResourceGraphBaseCmdlet" />
+    [Cmdlet(VerbsCommon.Search, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "Graph"), OutputType(typeof(PSObject))]
+    public class SearchAzureRmGraph : ResourceGraphBaseCmdlet
+    {
+        /// <summary>
+        /// The rows per page
+        /// </summary>
+        private const int RowsPerPage = 1000;
+
+        /// <summary>
+        /// Gets or sets the query.
+        /// </summary>s
+        [Parameter(Mandatory = true, Position = 0, ValueFromPipeline = true, HelpMessage = "Resource Graph query")]
+        [AllowEmptyString]
+        public string Query
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Gets or sets the subscriptions.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "List of subscriptions to run query against")]
+        public string[] Subscriptions
```

Comment:
@demyanenko please change this parameter name to `Subscription`.

From our [design guidelines](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/azure-powershell-design-guidelines.md#noun-singularity):

> _Avoid using plural names for parameters whose value is a single element. This includes parameters that take arrays or lists because the user might supply an array or list with only one element._

> _Plural parameter names should be used only in those cases where the value of the parameter is always a multiple-element value. In these cases, the cmdlet should verify that multiple elements are supplied, and the cmdlet should display a warning to the user if multiple elements are not supplied._